docs/readmes/bazel/agw_with_bazel.md

```diff
@@ -0,0 +1,573 @@
+- `bazel/external/name.BUILD`: `.BUILD` files for external repositories that are not bazelified.
+
+#### Starlark formatter
+
+To ensure the standardized formatting of all `BUILD.bazel` and `.bzl` files the Starlark formatter [buildifier](https://github.com/bazelbuild/buildtools/tree/master/buildifier) is used and run in [CI](#workflows).
+
+You can also run the Starlark format check locally, from your host or from any [Magma environment](#environments).
+
+To run the Starlark format check locally, run
+
+```bash
+cd $MAGMA_ROOT
+./bazel/scripts/run_buildifier.sh check
+```
+
+To run the Starlark formatter and to fix most errors that occur during the check, run
+
+```bash
+cd $MAGMA_ROOT
+./bazel/scripts/run_buildifier.sh format
+```
+
+### Environments
+
+The following Magma environments currently support Bazel:
+
+- Docker
+    - **devcontainer**
+        - The devcontainer is defined in `.devcontainer/Dockerfile` and meant for local development. The docker image is published at [`ghcr.io/magma/magma/devcontainer:latest`](https://github.com/magma/magma/pkgs/container/magma%2Fdevcontainer).
+    - **Bazel-base**
+        - The Bazel-base docker container is a minimal container where only required dependencies are installed. While this image is mainly intended to be used in CI, it can also be useful for local testing. The Bazel-base image is defined in `.devcontainer/bazel-base/Dockerfile` and published at [`ghcr.io/magma/magma/bazel-base:latest`](https://github.com/magma/magma/pkgs/container/magma%2Fbazel-base).
+- Vagrant VMs
+    - **magma-dev**
+        - The magma-dev virtual machine is intended for local development and for extended testing in CI. It can be used to run the Magma AGW services, as well as e.g. the [Python sudo tests](#python-sudo-tests).
+    - **magma-test**
+        - The magma-test virtual machine is used in CI to run the [LTE integration tests](#lte-integration-tests).
+
+### Caching
+
+Bazel has a very extensive and granular system of caching intermediate build outputs, artifacts and even test results. Builds that have many cache hits are much faster than builds without cache hits. The number of cache hits is reported at the end of each build, e.g. as `INFO: 4449 processes: 703 disk cache hit, 2377 internal, 1367 processwrapper-sandbox, 2 worker.`.
+
+Whenever cacheable results are produced a local cache entry is generated in the disk cache. Some of the cache locations can be found by running `bazel info` and looking e.g. at the `repository_cache` entry.
+
+> Info: A useful guide on [Bazel cache analysis](https://github.com/magma/magma/wiki/Bazel-cache-analysis) can be found on the GitHub Wiki pages.
+
+In CI caches have to be downloaded from elsewhere, this can be done either via [Docker images that contain pre-built caches](#docker-bazel-cache) or by using a [remote cache](#bazel-remote-cache).
```

Comment:
```suggestion
In CI, caches have to be downloaded from elsewhere, this can be done either via [Docker images that contain pre-built caches](#docker-bazel-cache) or by using a [remote cache](#bazel-remote-cache).
```